python/ray/data/read_api.py

```diff
@@ -242,6 +242,13 @@ def read_datasource(
             )
         )
 
+    if len(read_tasks) < parallelism:
+        logger.warning(
+            "The number of files for this dataset ({}) limits its parallelism to {}, "
+            "but the requested parallelism was {}. "
+            "To increase the available parallelism, use `ds.repartition()`.".format(
+                len(read_tasks), len(read_tasks), parallelism))
```

Comment:
I love this repartitioning suggestion, but I'm concerned that the `parallelism=200` default will cause this to be logged in the common case. Should we tweak the default parallelism to something smaller, or even something derived from the number of available CPU cores in the cluster?

Reply:
I changed this to only warning if parallelism is less than 50% of the available CPU cores, hopefully that will make it more actionable.